frontend/src/metabase/query_builder/components/filters/pickers/BooleanPicker/BooleanPickerRadio.tsx

```diff
@@ -1,48 +1,97 @@
-import _ from "underscore";
+import { useMemo } from "react";
+import { t } from "ttag";
+
+import * as ML from "metabase-lib";
+import { Radio } from "metabase/ui";
 
 import { useToggle } from "metabase/hooks/use-toggle";
+import { toLegacyFilter, getOperatorsMap } from "metabase-lib/compat";
 import type Filter from "metabase-lib/queries/structured/Filter";
 
-import { RadioContainer, Toggle, FilterRadio } from "./BooleanPicker.styled";
-
-import { OPTIONS, EXPANDED_OPTIONS } from "./constants";
-import { getValue } from "./utils";
+import { RadioContainer, Toggle } from "./BooleanPicker.styled";
+import { isBooleanValue } from "./utils";
 
 interface BooleanPickerProps {
   filter: Filter;
   onFilterChange: (filter: Filter) => void;
   className?: string;
 }
 
-function BooleanPicker({
+export function BooleanPickerRadio({
+  filter: legacyFilter,
   className,
-  filter,
   onFilterChange,
 }: BooleanPickerProps) {
-  const value = getValue(filter);
-  const [isExpanded, { toggle }] = useToggle(!_.isBoolean(value));
-
-  const updateFilter = (value: unknown) => {
-    if (_.isBoolean(value)) {
-      onFilterChange(filter.setOperator("=").setArguments([value]));
-    } else if (typeof value === "string") {
-      onFilterChange(filter.setOperator(value));
-    }
+  const { filterClause, query, legacyQuery, column, stageIndex } =
+    legacyFilter.getMLv2FilterClause();
+
+  const {
+    args: [value],
+    operator,
+  } = ML.filterParts(query, stageIndex, filterClause);
+
+  const operatorName = ML.displayInfo(query, stageIndex, operator)?.shortName;
+
+  const [isExpanded, { toggle }] = useToggle(!isBooleanValue(String(value)));
+
+  const operatorsMap = useMemo(
+    () => getOperatorsMap(query, stageIndex, column),
+    [query, stageIndex, column],
+  );
+
+  const filterOptions = useMemo(
+    () => [
+      {
+        name: t`True`,
+        value: "true",
+        filter: ML.filterClause(operatorsMap["="], column, [true]),
+      },
```

Comment:
Oh, that's nice ✨

Reply:
Wondering if we could use a clause as a value here, so we don't need to look it up later?

Reply:
the new radio component doesn't allow object values. At first I was thinking about extending the types, but I think the lookup is just as good, plus, matching objects to figure out which radio item is selected would be expensive and possibly error-prone.